src/gtk/textctrl.cpp

```diff
@@ -124,6 +124,56 @@ static void wxGtkTextApplyTagsFromAttr(GtkWidget *text,
         }
     }
 
+    if( attr.HasFontUnderlined() )
+    {
+        GtkTextTag *underlineColorTag;
+        PangoUnderline pangoUnderlineStyle;
+        switch( attr.GetUnderlineType() )
+        {
+            case wxTEXT_ATTR_UNDERLINE_NONE:
+                pangoUnderlineStyle = PANGO_UNDERLINE_NONE;
+                break;
+            case wxTEXT_ATTR_UNDERLINE_SOLID:
+                pangoUnderlineStyle = PANGO_UNDERLINE_SINGLE;
+                break;
+            case wxTEXT_ATTR_UNDERLINE_DOUBLE:
+                pangoUnderlineStyle = PANGO_UNDERLINE_DOUBLE;
+                break;
+            case wxTEXT_ATTR_UNDERLINE_WAVE:
+                pangoUnderlineStyle = PANGO_UNDERLINE_ERROR;
+                break;
+        }
+
+        g_snprintf(buf, sizeof(buf), "WXFONTUNDERLINEWITHEFFECT");
+        tag = gtk_text_tag_table_lookup( gtk_text_buffer_get_tag_table( text_buffer ),
+                                         buf );
+        if (!tag)
+        {
+            tag = gtk_text_buffer_create_tag( text_buffer, buf,
+                                              "underline-set", TRUE,
```

Comment:
Here and below is also capitalized `TRUE`.

Reply:
Thanks for the review, I agree with all the other comments (and there are also style issues such as writing `if(` instead of `if (`...), but not this one: we should indeed use `TRUE` in GTK code, this is what it uses and it's more better for consistency.